src/main/java/hudson/plugins/git/extensions/impl/SparseCheckoutPaths.java

```diff
@@ -36,27 +37,19 @@ public SparseCheckoutPaths(List<SparseCheckoutPath> sparseCheckoutPaths) {
 
     @Override
     public void decorateCloneCommand(GitSCM scm, Run<?, ?> build, GitClient git, TaskListener listener, CloneCommand cmd) throws IOException, InterruptedException, GitException {
-        if (! sparseCheckoutPaths.isEmpty()) {
+        if (!sparseCheckoutPaths.isEmpty()) {
             listener.getLogger().println("Using no checkout clone with sparse checkout.");
         }
     }
 
     @Override
     public void decorateCheckoutCommand(GitSCM scm, Run<?, ?> build, GitClient git, TaskListener listener, CheckoutCommand cmd) throws IOException, InterruptedException, GitException {
-        cmd.sparseCheckoutPaths(Lists.transform(sparseCheckoutPaths, SparseCheckoutPath.SPARSE_CHECKOUT_PATH_TO_PATH));
+        cmd.sparseCheckoutPaths(Lists.transform(sparseCheckoutPaths, new SparseCheckoutPathToPath(build.getEnvironment(listener))));
     }
 
     @Override
     public void determineSupportForJGit(GitSCM scm, @NonNull UnsupportedCommand cmd) {
-        cmd.sparseCheckoutPaths(Lists.transform(sparseCheckoutPaths, SparseCheckoutPath.SPARSE_CHECKOUT_PATH_TO_PATH));
-    }
-
-    @Extension
-    public static class DescriptorImpl extends GitSCMExtensionDescriptor {
-        @Override
-        public String getDisplayName() {
-            return "Sparse Checkout paths";
-        }
```

Comment:
Why moving so many methods outside the descriptor implementation class?  Seems like that will break many things and add unused public methods to the SparseCheckoutPaths object.

Reply:
Same